api/apps/geoprocessing/src/import/pieces-importers/project-metadata.piece-importer.ts

```diff
@@ -19,13 +19,75 @@ export class ProjectMetadataPieceImporter implements ImportPieceProcessor {
     private readonly fileRepository: FileRepository,
     @InjectEntityManager(geoprocessingConnections.apiDB)
     private readonly entityManager: EntityManager,
-  ) {}
+    private readonly logger: Logger,
+  ) {
+    this.logger.setContext(ProjectMetadataPieceImporter.name);
+  }
 
   isSupported(piece: ClonePiece): boolean {
     return piece === ClonePiece.ProjectMetadata;
   }
 
+  private async getRandomOrganizationId(): Promise<string> {
+    const [{ id }]: [{ id: string }] = await this.entityManager.query(`
+      SELECT id FROM organizations LIMIT 1
+    `);
+    return id;
+  }
+
   async run(input: ImportJobInput): Promise<ImportJobOutput> {
-    throw new Error('Missing implementation');
+    const { uris, resourceId } = input;
+
+    if (uris.length !== 1) {
+      const errorMessage = `uris array has an unexpected amount of elements: ${uris.length}`;
+      this.logger.error(errorMessage);
+      throw new Error(errorMessage);
+    }
+    const [projectMetadataLocation] = uris;
+
+    const readableOrError = await this.fileRepository.get(
+      projectMetadataLocation.uri,
+    );
+    if (isLeft(readableOrError)) {
+      const errorMessage = `Zip file not found: ${projectMetadataLocation.uri}`;
```

Comment:
very minor, but keep in mind that we'll store pieces uncompressed (whether in dev environments on local block storage, or in staging/prod/etc using Azure blob storage) - so I'd suggest to avoid specific file format lexicon... I'd just say something like `File with piece data for ${pieceKind}/${resourceId} is not available at ${projectMedatataLocation.uri}`.